src/Avalonia.Controls/Control.cs

```diff
@@ -347,7 +347,7 @@ internal void ScheduleOnLoadedCore()
         internal void OnLoadedCore()
         {
             if (_isLoaded == false &&
-                _isAttachedToVisualTree)
+                ((ILogical)this).IsAttachedToLogicalTree)
```

Comment:
Good catch, double checked WPF as well and it fires Loaded for logical not visual tree.